File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<Character>[] rowSet = new HashSet[9];
        HashSet<Character>[] colSet = new HashSet[9];
        HashSet<Character>[][] sqSet = new HashSet[3][3];
        for(int i = 0; i < 9; i++){
            for(int j = 0; j < 9; j++){
                char curr = board[i][j];
                if(curr == '.') continue;
                
                if(rowSet[i] == null){
                    rowSet[i] = new HashSet<Character>();
                }
                if(colSet[j] == null){
                    colSet[j] = new HashSet<Character>();
                }
                if(sqSet[i/3][j/3] == null){
                    sqSet[i/3][j/3] = new HashSet<Character>();
                }
                if(rowSet[i].contains(curr)
                  || colSet[j].contains(curr)
                  || sqSet[i/3][j/3].contains(curr)){
                    return false;
                }
                rowSet[i].add(curr);
                colSet[j].add(curr);
                sqSet[i/3][j/3].add(curr);
                
            }
        }
        return true;
    }
}

File: 0036-valid-sudoku/0036-valid-sudoku.kt
class Solution {
    fun isValidSudoku(board: Array<CharArray>): Boolean {
        val rowMaps = Array(9){ HashSet<Char>() };
        val colMaps = Array(9){ HashSet<Char>() };
        val sqMaps = Array(3) { Array(3) { HashSet<Char>() }};
        for(r in 0..8){
            for(c in 0..8){
                val curr = board[r][c]
                if(curr == '.') continue
                if(rowMaps[r].contains(curr) 
                   || colMaps[c].contains(curr)
                   || sqMaps[r/3][c/3].contains(curr)){
                    return false
                }
                rowMaps[r].add(curr)
                colMaps[c].add(curr)
                sqMaps[r/3][c/3].add(curr)
            }
        }
        return true
    }
}